File: Example/ObjcExample/ObjcExample/TableViewDemoController.m
//
//  TableViewDemoController.m
//  ObjcExample
//
//  Created by Animax Deng on 5/29/18.
//  Copyright © 2018 Animax. All rights reserved.
//

#import "TableViewDemoController.h"

@interface TableViewDemoController () <UITableViewDataSource, UITableViewDelegate, UITableViewDataSourcePrefetching>

@property (weak, nonatomic) IBOutlet UITableView *tableview;

@end

@implementation TableViewDemoController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.tableview.dataSource = self;
    self.tableview.delegate = self;
    self.tableview.prefetchDataSource = self;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return 10;
}
- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"cell"];
    return cell;
}

- (void)tableView:(UITableView *)tableView willDisplayFooterView:(UIView *)view forSection:(NSInteger)section {
    
}
- (void)tableView:(UITableView *)tableView willDisplayCell:(UITableViewCell *)cell forRowAtIndexPath:(NSIndexPath *)indexPath {
}

- (void)tableView:(UITableView *)tableView prefetchRowsAtIndexPaths:(NSArray<NSIndexPath *> *)indexPaths {
    NSLog(@"%@", indexPaths);
}


@end

File: Example/SwiftExample/SwiftExample/TableViewDemoController.swift
//
//  TableViewDemoController.swift
//  SwiftExample
//
//  Created by Animax Deng on 5/31/20.
//  Copyright © 2020 Animax. All rights reserved.
//

import UIKit

class TableViewDemoController: UIViewController, UITableViewDataSource, UITableViewDelegate {
    

    @IBOutlet weak var tableView: UITableView!
    
    override func viewDidLoad() {
        super.viewDidLoad()

        self.tableView.dataSource = self
        self.tableView.delegate = self
    }

    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        
        return 10
    }
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        
        let cell = tableView.dequeueReusableCell(withIdentifier: "cell")!
        return cell
    }
}
